includes/integrations/wc-blocks/class-orddd-lite-delivery-blocks.php

```diff
@@ -0,0 +1,337 @@
+			$time_slot = $data['orddd_lite_time_slot']; //phpcs:ignore
+
+			if ( has_filter( 'orddd_before_timeslot_update' ) ) {
+				$time_slot = apply_filters( 'orddd_before_timeslot_update', $time_slot );
+			}
+
+			$h_deliverydate = '';
+			if ( isset( $data['h_deliverydate'] ) ) { //phpcs:ignore
+				$h_deliverydate = $data['h_deliverydate']; //phpcs:ignore
+			}
+			if ( '' !== $time_slot && 'choose' !== $time_slot && 'NA' !== $time_slot && 'select' !== $time_slot ) {
+				if ( 'asap' === $time_slot ) {
+					Orddd_Lite_Common::update_order_meta( $order_id, $time_slot_label, esc_attr( __( 'As Soon As Possible.', 'order-delivery-date' ) ), $order );
+					Orddd_Lite_Common::update_order_meta( $order_id, '_orddd_time_slot', esc_attr( __( 'As Soon As Possible.', 'order-delivery-date' ) ), $order );
+				} else {
+					$order_time_slot = $time_slot;
+					$time_format     = get_option( 'orddd_lite_delivery_time_format' );
+					$time_slot_arr   = explode( ' - ', $time_slot );
+
+					if ( '1' === $time_format ) {
+						$from_time = date( 'H:i', strtotime( $time_slot_arr[0] ) ); //phpcs:ignore
+						if ( isset( $time_slot_arr[1] ) ) {
+							$to_time         = date( 'H:i', strtotime( $time_slot_arr[1] ) ); //phpcs:ignore
+							$order_time_slot = $from_time . ' - ' . $to_time;
+						} else {
+							$order_time_slot = $from_time;
+						}
+					} else {
+						$from_time = date( 'H:i', strtotime( $time_slot_arr[0] ) ); //phpcs:ignore
+					}
+					Orddd_Lite_Common::update_order_meta( $order_id, $time_slot_label, esc_attr( $time_slot ), $order );
+					Orddd_Lite_Common::update_order_meta( $order_id, '_orddd_time_slot', $order_time_slot, $order );
+
+					$delivery_date  = $h_deliverydate;
+					$delivery_date .= ' ' . $from_time;
+					$timestamp      = strtotime( $delivery_date );
+
+					Orddd_Lite_Common::update_order_meta( $order_id, '_orddd_lite_timeslot_timestamp', $timestamp, $order );
+
+					$total_fees = WC()->session->get( '_total_delivery_charges' );
+					if ( '' !== $total_fees && null !== $total_fees ) {
+						Orddd_Lite_Common::update_order_meta( $order_id, '_total_delivery_charges', $total_fees, $order );
+						WC()->session->__unset( '_total_delivery_charges' );
+					} else {
+						Orddd_Lite_Common::update_order_meta( $order_id, '_total_delivery_charges', '0', $order );
+					}
+
+					self::orddd_lite_update_lockout_timeslot( $h_deliverydate, $order_time_slot );
+				}
+				$order->save();
+			}
+
+			do_action( 'orddd_after_timeslot_update', $time_slot, $order_id );
+		}
+	}
+	/**
+	 * Update number of order for Delivery date and Time slot in options table
+	 *
+	 * @globals resource $wpdb WordPress Object
+	 *
+	 * @param string $delivery_date Selected Delivery date on the checkout page.
+	 * @param string $timeslot Selected time slot on the checkout page.
+	 *
+	 * @since 3.11.0
+	 */
+	public static function orddd_lite_update_lockout_timeslot( $delivery_date, $timeslot ) {
```

Comment:
This is already defined in class-orddd-lite-process.php. Please call the same function instead of creating it again.